### PR TITLE
EDA of Cor VS GO term and documents

### PR DESCRIPTION
**Pull request by zyhbetty**

### CODE REVIEW

File: R/GeneCorVSgotermEDA.R [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  # Calculate pearson correlation between gene expression
+  # profiles A and B in prf identified by the gene symbol.
+  # A and B can be either gene symbol or index.
+
+  r <- cor(prf[A, ], prf[B, ], use = "pairwise.complete.obs")
+  return(r)
+}
+
+
+# ====  PACKAGES  ==============================================================
```
Comment (rn1n3r) [inline R/GeneCorVSgotermEDA.R]: You can remove this as the packages should be installed when you list them as Imports in DESCRIPTION.

File: R/GeneCorVSgotermEDA.R [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+#BiocManager::install("GOSemSim", version = "3.8")
+#BiocManager::install("org.Hs.eg.db", version = "3.8")
+#install.packages("ggcorrplot")
+# packages for calculating GO similarity
+
+
+
+
+#'
+#' \code{myPlot()}
```
Comment (rn1n3r) [inline R/GeneCorVSgotermEDA.R]: I think you can make the name of your function more expressive (i.e. specify what kind of plots you're making)

File: R/GeneCorVSgotermEDA.R [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+#' Investigates relationship of expression correlation and semantic similarity of input genes
+#' and produces plots and summary of model info
+#'
+#' @param geneSet A set of genes of interest to investigate
+#' @return A heatmap of co-expression correlation of each pair of genes in the geneSet,
```
Comment (rn1n3r) [inline R/GeneCorVSgotermEDA.R]: The name of the returned variable must directly follow `@return` - e.g. `@return myData`

File: R/GeneCorVSgotermEDA.R [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    #     Collecting gene correlations and GO semantic similarity and build relationship model
+    #     for each pair of genes to prove the hypothesis
+    # Parameters:
+    #     A : a set of genes of interets
+    # Value:
+    #     result: summary of the model withe scatterplot.
+
+    # code ...
+    # Loading necessary data
+    myURL <- paste0("http://steipe.biochemistry.utoronto.ca/abc/assets/",
```
Comment (rn1n3r) [inline R/GeneCorVSgotermEDA.R]: Please use fetchData() to load the GEO and HGNC data

File: R/GeneCorVSgotermEDA.R [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+        }
+        GeneA <- c(GeneA, as.character(a))
+        GeneB <- c(GeneB,as.character(b))
+        CorGene <- c(CorGene, cor)
+        scoGO <- c(scoGO,as.numeric(scoreGO))
+      }
+    }
+
+    mydata <- data.frame(GeneA, GeneB, CorGene,scoGO)
```
Comment (rn1n3r) [inline R/GeneCorVSgotermEDA.R]: Consider renaming this variable to something that makes it more clear what it is storing.

**Review by rn1n3r — DISMISSED** [review]
Remember to add yourself as an author to the DESCRIPTION file.

File: R/GeneCorVSgotermEDA.R (hunk 2 above)
Comment (zyhbetty) [reply R/GeneCorVSgotermEDA.R]: fixed to myScatterPlot()

File: R/GeneCorVSgotermEDA.R (hunk 3 above)
Comment (zyhbetty) [reply R/GeneCorVSgotermEDA.R]: fixed to @return (list) and will return list of graphs

File: DESCRIPTION [hunk 6]
```diff
@@ -27,7 +31,11 @@ Imports:
     cluster,
     grid,
     igraph,
-    VennDiagram
+    VennDiagram,
+    GOSemSim,
+    ggcorrplot,
+    ggplot2,
+    org.Hs.eg.db
```
Comment (hyginn) [inline DESCRIPTION]: **We won't merge that in.** I'll comment on the mailing list. You are not the only one to try this.

**Review by hyginn — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 6 above)
Comment (zyhbetty) [reply DESCRIPTION]: I have added new functions and files that will not merge org.Hs.eg.db

File: R/GeneCorVSgotermEDA.R [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+#' @return (NULL)
+#' #' @author {Yuhan Zhang} (aut)
+#' @examples
+#' \dontrun{
+#' installHumanGenomeAnnotation()
+#' }
+#' @export
+installHumanGenomeAnnotation <- function() {
+if (!requireNamespace("BiocManager", quietly = TRUE)) {
```
Comment (hyginn) [inline R/GeneCorVSgotermEDA.R]: Unneccessary: `BiocManager` is in `Imports:`

File: R/GeneCorVSgotermEDA.R [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+
+    len <- length(geneSet)
+    for (i in seq_along(geneSet)){
+      for (j in i : len){
+        a <- geneSet[i]
+        b <- geneSet[j]
+        if (i != j){ # different gene
+
+         cor <- as.numeric(corGenes(a,b,myQNXP))
```
Comment (hyginn) [inline R/GeneCorVSgotermEDA.R:75]: You don't need this function. It's just one expression, which you can just use in your code.

**Review by hyginn — COMMENTED** [review]
Vignette does not explain anything. Code needs some fixes, but is basically sound; review comments were addressed. Will make two local changes, resolve conflicts and merge.

**Comment by hyginn** [comment]
Waiting for Travis - will merge after check passes.

**Review by hyginn — APPROVED** [review]

**Review by hyginn — COMMENTED** [review]
This took considerable changes to bring into mergeable shape and I suspect it never did pass local check without error, warnings, or notes.

**Comment by hyginn** [comment]
Couldn't get Travis to pass - merged functions manually, required significant changes: probably never passed locally anyway. Vignette not worth merging.  Evaluated as-is. Closed.

**Pull request closed without merging**